diffrax/_integrate.py

```diff
@@ -405,7 +406,7 @@ def _body_fun(_save_state):
                 _ts = _save_state.ts.at[_save_state.save_index].set(_t)
                 _ys = jtu.tree_map(
                     lambda __y, __ys: __ys.at[_save_state.save_index].set(__y),
-                    fn(_t, _y, args),
+                    fn(direction * _t, _y, args),
```

Comment:
I think this change needs to include in all the other places we call `fn`, i.e. when saving at `t0`, `t1`, `steps`, dense`.

I think it'd better to simply wrap `.fn` inside of `saveat`, during the initial set-up for the solve. It's a bit of a mouthful, but something like this:
```python
def _subsaveat_direction_fn(x):
    if _is_subsaveat(x):
        direction_fn = lambda t, y, args: x.fn(direction * t, y, args)
        return eqx.tree_at(lambda x: x.fn, x, direction_fn)
    else:
        return x

saveat = jtu.tree_map(_subsaveat_direction, saveat, is_leaf=_is_subsaveat)
```

Reply:
That's actually what I had originally, but I thought direction would then get closured in and cause issues. I will roll back to that version